oauth2_provider/locale/fr/LC_MESSAGES/django.po

```diff
@@ -0,0 +1,193 @@
+msgstr "Le token d'accès a expiré."
+
+#: oauth2_validators.py:225
+msgid "The access token is valid but does not have enough scope."
+msgstr "Le token d'accès est valide, mais sa portée n'est pas suffisante."
+
+#: templates/oauth2_provider/application_confirm_delete.html:6
+msgid "Are you sure to delete the application"
+msgstr "Êtes-vous sûr de vouloir supprimer l'application"
+
+#: templates/oauth2_provider/application_confirm_delete.html:12
+#: templates/oauth2_provider/authorize.html:29
+msgid "Cancel"
+msgstr "Annuler"
+
+#: templates/oauth2_provider/application_confirm_delete.html:13
+#: templates/oauth2_provider/application_detail.html:38
+#: templates/oauth2_provider/authorized-token-delete.html:7
+msgid "Delete"
+msgstr "Supprimer"
+
+#: templates/oauth2_provider/application_detail.html:10
+msgid "Client id"
+msgstr "ID du client"
+
+#: templates/oauth2_provider/application_detail.html:15
+msgid "Client secret"
+msgstr "Secret client"
+
+#: templates/oauth2_provider/application_detail.html:20
+msgid "Client type"
+msgstr "Type de client"
+
+#: templates/oauth2_provider/application_detail.html:25
+msgid "Authorization Grant Type"
+msgstr "Type de flux d'autorisation"
+
+#: templates/oauth2_provider/application_detail.html:30
+msgid "Redirect Uris"
+msgstr "URIs de redirection"
+
+#: templates/oauth2_provider/application_detail.html:36
+#: templates/oauth2_provider/application_form.html:35
+msgid "Go Back"
+msgstr "Revenir en arrière"
+
+#: templates/oauth2_provider/application_detail.html:37
+msgid "Edit"
+msgstr "Modifier"
+
+#: templates/oauth2_provider/application_form.html:9
+msgid "Edit application"
+msgstr "Modifier l'application"
+
+#: templates/oauth2_provider/application_form.html:37
+msgid "Save"
+msgstr "Sauvegarder"
+
+#: templates/oauth2_provider/application_list.html:6
+msgid "Your applications"
+msgstr "Vos applications"
+
+#: templates/oauth2_provider/application_list.html:14
+msgid "New Application"
+msgstr "Nouvelle application"
+
+#: templates/oauth2_provider/application_list.html:17
+msgid "No applications defined"
+msgstr "Pas d'applications définies"
+
+#: templates/oauth2_provider/application_list.html:17
+msgid "Click here"
+msgstr "Cliquez ici"
+
+#: templates/oauth2_provider/application_list.html:17
+msgid "if you want to register a new one"
+msgstr "si vous voulez en enregistrer une nouvelle"
+
+#: templates/oauth2_provider/application_registration_form.html:5
+msgid "Register a new application"
+msgstr "Enregistrer une application"
+
+#: templates/oauth2_provider/authorize.html:8
+#: templates/oauth2_provider/authorize.html:30
+msgid "Authorize"
+msgstr "Autoriser"
+
+#: templates/oauth2_provider/authorize.html:17
+msgid "Application requires the following permissions"
+msgstr "L'application nécessite les permissions suivantes"
+
+#: templates/oauth2_provider/authorized-token-delete.html:6
+msgid "Are you sure you want to delete this token?"
+msgstr "Êtes-vous sûr de vouloir supprimer ce jeton ?"
```

Comment:
This might seem weird but in French there is a space before `?`

Reply:
[Oui !](https://mashable.com/article/leaving-space-before-end-punctuation-texting#:~:text=French.,have%20a%20space%20preceding%20them.)